### DISCUSSION TITLE
[Idea] Native support of backoffice restriction by ip address

### DISCUSSION THREAD

**Discussion by jf-viguier** [PrestaShop/PrestaShop#30855] [Ideas]
or ip range at htaccess level

**Comment by hrostyk** [comment]
In my opinion it is better to do it on the web-server conf or htaccess.

**Reply by jf-viguier** [reply]
it's the idea : an UI to generate the htaccess rules

**Comment by prestaalba** [comment]
You can do it using a firewall rule on your server or for free on Cloudflare.

**Comment by ytilotti** [comment]
A restriction IP is unsecure. So why add it?

**Reply by jf-viguier** [reply]
argument please ?

**Reply by ytilotti** [reply]
An IP restriction is not basically secure and bypassable. For a demo of Front-office is not critical but for an admin access maybe. An htpasswd via login/pass (strong) is more logical. 

OWASP says:
```
Do not rely on falsifiable information as a unique authentication method such as IP address or address range masks, DNS or Reverse DNS, Referrer headers
```

**Reply by touchweb-vincent** [reply]
For our target threats, it fits the needs as "a complementary authentification method".

As OWAPS said : "Do not rely on falsifiable information as a *UNIQUE* authentication method". As long as you use it as complementary and not unique, there is no problem with this.

**Comment by eternoendless** [comment]
I suppose this could be implemented the same way as the "maintenance IP" feature.
